backtracking/rat_in_maze.py

```diff
@@ -0,0 +1,167 @@
+    1 are open paths we can use.
+    Parameters :
+        maze(2D matrix) : maze
+        size : size of our maze(square)
+    Returns:
+        This method returns true and array if solution found otherwise false and None
+
+    >>> maze = [[0, 1, 0, 1, 1],
+    ...         [0, 0, 0, 0, 0],
+    ...         [1, 0, 1, 0, 1],
+    ...         [0, 0, 1, 0, 0],
+    ...         [1, 0, 0, 1, 0]]
+    >>> solveMaze(maze)
+    [1, 0, 0, 0, 0]
+    [1, 1, 1, 1, 0]
+    [0, 0, 0, 1, 0]
+    [0, 0, 0, 1, 1]
+    [0, 0, 0, 0, 1]
+
+    >>> maze = [[0, 1, 0, 1, 1],
+    ...         [0, 0, 0, 0, 0],
+    ...         [0, 0, 0, 0, 1],
+    ...         [0, 0, 0, 0, 0],
+    ...         [0, 0, 0, 0, 0]]
+    >>> solveMaze(maze)
+    [1, 0, 0, 0, 0]
+    [1, 0, 0, 0, 0]
+    [1, 0, 0, 0, 0]
+    [1, 0, 0, 0, 0]
+    [1, 1, 1, 1, 1]
+
+    >>> maze = [[0, 0, 0],
+    ...         [0, 1, 0],
+    ...         [1, 0, 0]]
+    >>> solveMaze(maze)
+    [1, 1, 1]
+    [0, 0, 1]
+    [0, 0, 1]
+
+    >>> maze = [[0, 1, 0],
+    ...         [0, 1, 0],
+    ...         [1, 0, 0]]
+    >>> solveMaze(maze)
+    Solution does not exists!
+
+    >>> maze = [[0, 1],
+    ...         [1, 0]]
+    >>> solveMaze(maze)
+    Solution does not exists!
+    """
+    size = len(maze)
+    # We need to create solution object to save path.
+    solutions = [[0 for _ in range(size)] for _ in range(size)]
+    solved = runmaze(maze, 0, 0, solutions)
+
+    if solved:
+        result = "".join(str(row) + "\n" for row in solutions)
+        print(result[: len(result) - 1])  # For last \n
+    else:
+        print("Solution does not exists!")
+
+
+"""
+This method is recursive method which starts from i and j
+and goes with 4 direction option up, down, left, right
+if path found to destination it breaks and return True
+otherwise False
+Parameters:
+    maze(2D matrix) : maze
+    size : size of our maze(square)
+    i, j : coordinates of matrix
+    solutions(2D matrix) : solutions
+Returns:
+    Boolean if path is found True, Otherwise False.
+
+    >>> maze = [[0, 1, 0, 1, 1],
+    ...         [0, 0, 0, 0, 0],
+    ...         [1, 0, 1, 0, 1],
+    ...         [0, 0, 1, 0, 0],
+    ...         [1, 0, 0, 1, 0]]
+    >>> solveMaze(maze)
+    True
+
+    >>> maze = [[0, 1, 0, 1, 1],
+    ...         [0, 0, 0, 0, 0],
+    ...         [0, 0, 0, 0, 1],
+    ...         [0, 0, 0, 0, 0],
+    ...         [0, 0, 0, 0, 0]]
+    >>> solveMaze(maze)
+    True
+
+    >>> maze = [[0, 0, 0],
+    ...         [0, 1, 0],
+    ...         [1, 0, 0]]
+    >>> solveMaze(maze)
+    True
+
+    >>> maze = [[0, 1, 0],
+    ...         [0, 1, 0],
+    ...         [1, 0, 0]]
+    >>> solveMaze(maze)
+    False
+
+    >>> maze = [[0, 1],
+    ...         [1, 0]]
+    >>> solveMaze(maze)
+    False
+"""
+
+
+def runmaze(maze, i, j, solutions):
+    size = len(maze)
+    # Final check point.
+    if (i == (size - 1)) and (j == (size - 1)):
+        fill_matrix(solutions, i, j)
```

Comment:
```suggestion
        solutions[i, j] = 1
```
Faster and more intuitive.